src/Guard.php

```diff
@@ -88,19 +88,7 @@ public function __construct(
             throw new RuntimeException('CSRF middleware failed. Minimum strength is 16.');
         }
         $this->strength = $strength;
-        if (is_array($storage)) {
-            $this->storage = &$storage;
-        } elseif ($storage instanceof ArrayAccess) {
-            $this->storage = $storage;
-        } else {
-            if (!isset($_SESSION)) {
-                throw new RuntimeException('CSRF middleware failed. Session not found.');
-            }
-            if (!array_key_exists($prefix, $_SESSION)) {
-                $_SESSION[$prefix] = [];
-            }
-            $this->storage = &$_SESSION[$prefix];
-        }
+        $this->storage = &$storage;
```

Comment:
I'm not 100% sure that we want to do this if `$storage` is an instance of `ArrayAccess` as it's already passed by reference.